crates/matrix-sdk-ui/src/timeline/controller/state_transaction.rs

```diff
@@ -429,16 +434,12 @@ impl<'a> TimelineStateTransaction<'a> {
             }
 
             TimelineFocusKind::Live => {
-                // The live timeline doesn't apply any additional
-                // filtering: the event *should* be added!
-                true
+                return thread_root.is_none();
             }
 
-            TimelineFocusKind::Thread => {
-                // The thread timeline doesn't apply any additional
-                // for now. It will however do so in the future, as
-                // will the live one
-                true
+            TimelineFocusKind::Thread { root_event_id } => {
+                event.event_id() == root_event_id
+                    || thread_root.as_ref().is_some_and(|r| r == root_event_id)
```

Comment:
Unfortunately, I think this is slightly incorrect in the presence of aggregations applying to in-thread events, but we can deal with this later.

Reply:
Or maybe not! aggregations don't care about the result of `should_add_event_item` \o/